src/System/Wlog/Exception.hs

```diff
@@ -0,0 +1,52 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE ExplicitForAll      #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- | This module deals with Exception logging.
+
+module System.Wlog.Exception
+       ( logException
+       , catchLog
+       ) where
+
+import Universum
+
+import Control.Exception.Base (Exception)
+
+import System.Wlog.CanLog (WithLogger, WithLoggerIO, logError)
+
+-- | Logs exception's description with ''System.Wlog.Severity.Error' 'System.Wlog.Severity.Severity'
+logException :: forall e m . (WithLogger m, Exception e) => e -> m ()
+logException = logError . show
+
+{- | Runs the action, if an exception is raised the 'logException' is executed.
+
+==== __Example__
+Here is very simple example of usage 'catchLog' on IO functions:
+
+@
+main :: IO ()
+main = do
+    buildAndSetupYamlLogging productionB "log-config.yaml"
+    usingLoggerName "new-logger" runWithExceptionLog
+
+runWithExceptionLog :: (WithLoggerIO m, MonadCatch m) => m ()
+runWithExceptionLog = catchLog @IOException (liftIO simpleIOfun)
+
+simpleIOfun :: IO ()
+simpleIOfun = getLine >>= readFile >>= putStrLn
+
+@
+
+and when run you will get:
+
+>>> stack exec except-log
```

Comment:
You're running this example with `stack exec except-log` command, but there's no `except-log` target in this project. You can either add this target to project or use some abstract generic command in CLI like `run-main-from-this-example`.